WordPress/wordpress-playground
# Blueprints: Rename importFile to importWxr, switch to humanmade/WordPress importer

## Problem addressed

The importFile step works by interacting with HTML form like a user would. Problem is, the WXR importer requires some interactions with the forms which we fake by adjusting the POST data before submitting the form. This approach causes a number of issues listed below.

More context: https://github.com/WordPress/wordpress-playground/issues/1183.

## Implementation

Refactors the importFile Blueprint step to:

* Be named importWxz. The old name continues to work, though.
* Use the humanmade/WordPress-Importer for importing content – it is
  more reliable than the official wordpress-importer and has a
  convenient PHP API.
* Drop WXZ support – it's a maintenance burden and doesn't add clear
  value.
* Remove dependency on DOMParser – this step should now work in `wp-now`

## Testing instructions

1. Go to http://localhost:5400/website-server/?import-wxr=https://raw.githubusercontent.com/helgatheviking/Simple-User-Listing/3e38ea3eb3a057424fbae4305cba2fd9f73d896b/demo-content/demo-content.xml&php=8.0&wp=6.5&storage=none&php-extension-bundle=kitchen-sink
2. Confirm the "Simple User Listing Block" page was created
3. Confirm in wp-admin a bunch of user accounts were created
4. Confirm in wp-admin the WordPress Importer plugin was installed

Closes https://github.com/WordPress/wordpress-playground/issues/1183
Closes https://github.com/WordPress/wordpress-playground/issues/379
Closes https://github.com/WordPress/wordpress-playground/issues/1158
Closes https://github.com/WordPress/wordpress-playground/issues/1161

cc @dmsnell
